client/src/features/ProjectPageV2/ProjectPageContent/Settings/ProjectSettings.tsx

```diff
@@ -15,10 +15,120 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-export default function ProjectPageSettings({
-  projectId,
-}: {
-  projectId: string;
-}) {
-  return <small>Settings {projectId}</small>;
+import cx from "classnames";
+import { useCallback } from "react";
+import { ArrowClockwise } from "react-bootstrap-icons";
+import { useForm } from "react-hook-form";
+import { Button, Form } from "reactstrap";
+import { SuccessAlert } from "../../../../components/Alert.jsx";
+import { Loader } from "../../../../components/Loader.tsx";
+import { RtkErrorAlert } from "../../../../components/errors/RtkErrorAlert.tsx";
+import { Project } from "../../../projectsV2/api/projectV2.api.ts";
+import { usePatchProjectsByProjectIdMutation } from "../../../projectsV2/api/projectV2.enhanced-api.ts";
+import ProjectDescriptionFormField from "../../../projectsV2/fields/ProjectDescriptionFormField.tsx";
+import ProjectNameFormField from "../../../projectsV2/fields/ProjectNameFormField.tsx";
+import ProjectNamespaceFormField from "../../../projectsV2/fields/ProjectNamespaceFormField.tsx";
+import ProjectVisibilityFormField from "../../../projectsV2/fields/ProjectVisibilityFormField.tsx";
+import { ProjectV2Metadata } from "../../../projectsV2/show/ProjectV2EditForm.tsx";
+import ProjectPageDelete from "./ProjectDelete.tsx";
+interface ProjectSettingsFormProps {
+  project: Project;
+}
+export function ProjectSettingsForm({ project }: ProjectSettingsFormProps) {
+  const {
+    control,
+    formState: { errors },
```

Comment:
You can also subscribe to `isDirty` here to make the update button active only when something has been changed.

```suggestion
    formState: { errors, isDirty },
```